mcp/mcp-cli-api/src/main/java/software/amazon/smithy/java/mcp/cli/SmithyMcpCommand.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.mcp.cli;
+
+import static software.amazon.smithy.java.mcp.cli.ConfigUtils.loadOrCreateConfig;
+
+import java.util.concurrent.Callable;
+import software.amazon.smithy.java.logging.InternalLogger;
+import software.amazon.smithy.java.mcp.cli.model.Config;
+
+/**
+ * Base class for all Smithy MCP CLI commands.
+ * <p>
+ * This class implements the Callable interface to provide a consistent execution pattern
+ * for all MCP CLI commands. It handles loading the configuration, executing the command,
+ * and providing appropriate error handling.
+ */
+public abstract class SmithyMcpCommand implements Callable<Integer> {
+
+    InternalLogger LOG = InternalLogger.getLogger(SmithyMcpCommand.class);
+
+    @Override
+    public final Integer call() throws Exception {
+        try {
+            var config = loadOrCreateConfig();
+            execute(config);
+            return 0;
+        } catch (IllegalArgumentException e) {
+            System.out.println("Invalid input : [" + e.getMessage() + "]");
```

Comment:
* Why println here and LOG in the Exception arm?
* Please don't discard the stack trace. Do `e.printStackTrace(System.out)` if you must, although we should likely be using the logger here (unless you intended for this to go to the mcp client's logs, in which case stderr is a better choice)

Reply:
So the idea was we only emit messages for errors and only print stack traces. I didn't want to log a stack trace for wrong inputs.

I think I need to do a second pass on logging, and also configure the logger. Will cleanup logging and println in the CLIs in a followup.